test-apps/appui-test-app/standalone/src/frontend/appui/frontstages/TestPopoutFrontstage.tsx

```diff
@@ -48,3 +52,32 @@ export const createTestPopoutFrontstage = () => {
   `);
   document.adoptedStyleSheets.push(sheet);
 })();
+
+function FixedProgressRadial(
+  props: React.ComponentProps<typeof ProgressRadial>
+) {
+  const ref = React.useRef<HTMLDivElement>(null);
+  const [key, setKey] = React.useState(0);
+  React.useEffect(() => {
+    if (!ref.current) return;
+    const themeProvider = ref.current.closest(
+      ".uifw-preview-reparentPopoutWidgets-popoutThemeProvider"
+    );
+    if (!themeProvider) return;
+    const listener = () => {
+      Logger.logInfo(
+        SampleAppIModelApp.loggerCategory(FixedProgressRadial),
+        "reparented"
+      );
+      // For now we just force a re-mount to copy styles.
+      setKey((prev) => prev + 1);
+    };
+
+    // Event is not dispatched to every node in the widget element tree, so we need to listen on the theme provider.
+    themeProvider.addEventListener("appui:reparent", listener);
+    return () => {
+      themeProvider.removeEventListener("appui:reparent", listener);
+    };
```

Comment:
The event is emitted after the element is added to a new parent, wouldn't this mean that this listener would not be called since the event would bubble up to the "new document"?

Reply:
I agree about the internal classes and I've added `FixedProgressRadial` only temporarily until this is fixed by iTwinUI.

Reply:
I expect that this test is meant to be representative of how we would handle the event in iTwinUI (and maybe also in other libraries like kiwi), so it should not use these internal classes.

Maybe the event should be emitted *before* the element is added to the new parent, so that we can listen to it in the original document? Or maybe we need two events like `appui:reparentstart` and `appui:reparentend`?

Alternatively, I wonder if there is a way to listen to this event during the capture phase so that we don't need to reach for `document`. 🤔

Reply:
Emitting two events might work, but the usage would be very complicated since consumers would need to add/remove `appui:reparentend` listener in `appui:reparentstart`. Additionally, since the widget node is removed from the DOM (no `parentElement`, only `ownerDocument` is defined), we'd need to dispatch `appui:reparentstart` to `ownerDocument`, so you'd need to re-add `appui:reparentstart` from `appui:reparented` as well.

What if we dispatch the event to the global `window` object and use custom event `detail` instead of event `target` to specify the reparented node? I've pushed some code to showcase that.

Reply:
This looks much better! I think it might actually work 👍